45823 - Appointment List - Mobile View v2

vbahinwillit opened the pull request:
## Summary
Mobile view for the VAOS Homepage appointment list redesign

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#45823

## Testing done
Unit testing and e2e testing

## Screenshots

## What areas of the site does it impact?
VAOS

## Tasks
- [x] #22899 
- [x] #22890 
- [x] #23008 
- [x] #23059 
- [x] #23007 

## Acceptance criteria
- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

vbahinwillit committed:
rename 'Grid' component to 'AppointmentFlexGrid'

vbahinwillit committed:
fix issue with phone-alt icon not displaying

vbahinwillit committed:
move hooks to hooks folder folder

vbahinwillit committed:
fix text shifting on hover and when active
fix border
add style for truncation
modified column width

vbahinwillit committed:
add new grid components

vbahinwillit committed:
add new appointment model

vbahinwillit committed:
modified to use click and keydown hooks

vbahinwillit committed:
modified interface

vbahinwillit committed:
modified to return hash table

vbahinwillit committed:
modified to work with hash table

vbahinwillit committed:
refactored to use new flex grid components

vbahinwillit committed:
fix failing unit test due to missing code

vbahinwillit committed:
fix issue with no appointments displaying when
appointments exisits

vbahinwillit committed:
fix 'getTypeOfCareName'

vbahinwillit committed:
fix 'getTypeOfCareName'

vbahinwillit committed:
refactored 'rederFlexGrid' to component

vbahinwillit committed:
add unstyled list class

jenniemc reviewed (changes requested):
For this iteration, I only reviewed the files supporting `UpcomingAppointmentLayout` component. So far it's looking real good in the desktop and mobile small-screen version. I included some suggestions for accessibility.

You may want to check out the text alignment on the print version.
<img width="561" alt="Screenshot 2023-01-11 at 3 08 13 PM" src="https://user-images.githubusercontent.com/54327023/211941250-179edcd4-e50d-48e5-903f-1396ed96c2f2.png">

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2

rmessina1010 reviewed (approved):
LGTM

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2

vbahinwillit committed:
fix failing unti tests

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2

vbahinwillit commented on src/applications/vaos/appointment-list/models/AppointmentModel.js:
```diff
@@ -0,0 +1,170 @@
+    return getAppointmentTimezone(this.data);
+  }
+
+  getAppointmentDetails() {
```
Thanks for the suggestiong. Changed to 'getAppointmentLocality'.

vbahinwillit commented on src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:
```diff
@@ -0,0 +1,142 @@
+              size="1"
+              style={{ minWidth: '25px', maxWidth: '25px' }}
+            >
+              <span>{model.getStartDate().format('ddd')}</span>
+              <span className="sr-only"> {model.getTimeZoneAbbr()}</span>
```
Thanks for the suggestion. I will look into what Voice Over announces when we start 508 testing. The announcement of timezone was in the original code though.

vbahinwillit committed:
fix truncation/alignment issue

vbahinwillit committed:
removed model concept per code review

vbahinwillit committed:
fix heading level
fix spacing

vbahinwillit committed:
update text to match copy doc
various cosmetic changes
rename RequestListItemGroup to RequestAppointmentLayout

vbahinwillit committed:
fix error

vbahinwillit committed:
fix failing tests

jenniemc reviewed (approved):
LGTM!!

vbahinwillit committed:
Merge branch 'main' into vbahinwillit/issue45823-v2
